drivers/lis2dh12/lis2dh12.c

```diff
@@ -184,9 +184,13 @@ int lis2dh12_init(lis2dh12_t *dev, const lis2dh12_params_t *params)
     /* enable all axes, set sampling rate and scale */
     LIS2DH12_CTRL_REG1_t reg1 = {0};
 
+    /* cppcheck-suppress unreadVariable; (reason: union type is used for read) */
```

Comment:
@maribu Here we have no volatile variable definitions, but `cppcheck` complains anyway. So it looks like `cppcheck` does not recognize the type `unit` after all. Missing header files are probably not the problem in this case.

Reply:
@maribu I played a bit with `cppcheck` and the single file that made the problems using command:
```
cppcheck --std=c99 --enable=style --force --error-exitcode=2 --quiet -j 1 --inline-suppr \
--suppress=unusedStructMember --template '{file}:{line}: {severity} ({id}): {message}' \
drivers/lis2dh12/lis2dh12.c
```
`cppcheck` version 1.82 (local Ubuntu 18.04 LTS version) doesn't show any warnings.
`cppcheck` version 1.90 in RIOT docker shows these 32 warnings as in CI.
`cppcheck` version 2.6 (newest version from `sourgeforge.net`) shows only 5 warnings.

Adding the required include pathes
```
-I drivers/include -I drivers/lis2dh12/include/
```
doesn't help with `cppcheck` version 1.90 but solves all warnings with `cppcheck` version 2.6.

So it seems that we need correct include pathes as well as a newer `cppcheck` version in CI.